nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -214,16 +237,154 @@ def reset_tools(self):
         else:
             self.data_collector.reset()
         if self.metrics_calculator is None:
-            self.metrics_calculator = NormMetricsCalculator()
+            self.metrics_calculator = NormMetricsCalculator(block_sparse_size=self.block_sparse_size)
```

Comment:
what happened if weight shape cannot be devided by block size?

Reply:
will raise mask shape mismatch exception